src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -590,15 +590,15 @@ private static ExpressionSyntax SkipParensAndNullSuppressions(ExpressionSyntax e
                     if (MessageID.IDS_FeatureSpanCharConstantPattern.CheckFeatureAvailability(diagnostics, Compilation, node.Location))
                     {
                         // report missing member and use site diagnostics
-                        if (inputType.IsReadOnlySpanChar())
-                        {
-                            _ = GetWellKnownTypeMember(WellKnownMember.System_MemoryExtensions__SequenceEqual_ReadOnlySpan_T, diagnostics, node);
-                        }
-                        else
-                        {
-                            _ = GetWellKnownTypeMember(WellKnownMember.System_MemoryExtensions__SequenceEqual_Span_T, diagnostics, node);
-                        }
-                        _ = GetWellKnownTypeMember(WellKnownMember.System_MemoryExtensions__AsSpanString, diagnostics, node);
+                        bool isReadOnlySpan = inputType.IsReadOnlySpanChar();
+                        _ = GetWellKnownTypeMember(
+                            isReadOnlySpan ? WellKnownMember.System_MemoryExtensions__SequenceEqual_ReadOnlySpan_T : WellKnownMember.System_MemoryExtensions__SequenceEqual_Span_T,
+                            diagnostics,
+                            syntax: node);
+                        _ = GetWellKnownTypeMember(WellKnownMember.System_MemoryExtensions__AsSpan_String, diagnostics, syntax: node);
```

Comment:
Do we have a test where `System_MemoryExtensions__AsSpan_String` is missing? Same question for the other types.